### PR TITLE
Add metric for ControlPlaneMachineSet state

### PR DESCRIPTION
**Pull request by RaphaelBut**
Adds a new metric 'cpms_enabled' to indicate the state of the controlplane machine set on an osd cluster.

The metric is set to 1 if cpms is in state 'Active' and '0' otherwise.
```
# HELP cpms_enabled Indicates if the controlplanemachineset is enabled
# TYPE cpms_enabled gauge
cpms_enabled{_id="cluster-id",name="osd_exporter"} 1
```

It's part of the reenablement of CPMS https://issues.redhat.com/browse/OSD-20588 
Tested locally against a cluster.

### CODE REVIEW

File: controllers/cpms/cpms_controller.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2022.
```
Comment (typeid) [inline controllers/cpms/cpms_controller.go:2]: Should this be updated at some point?  I suppose it's autogenerated with 2022?

**Review by typeid — APPROVED** [review]
Great work @RaphaelBut!
/lgtm

File: controllers/cpms/cpms_controller.go (hunk 1 above)
Comment (RaphaelBut) [reply controllers/cpms/cpms_controller.go:2]: ah yes. I just copied it from another file. Let me update to 2024

**Pull request merged**